projects/OG-Analytics/src/main/java/com/opengamma/analytics/financial/provider/calculator/discounting/CrossGammaSingleCurveCalculator.java

```diff
@@ -81,7 +81,7 @@ public DoubleMatrix2D calculateCrossGamma(final InstrumentDerivative instrument,
     ArgumentChecker.isTrue(multicurve.getAllNames().size() == 1,
         "provider should have only one curve for GammaSingleCurve computation");
     String name = multicurve.getAllNames().iterator().next();
-    Currency ccy = multicurve.getCurrencyForName(name);
+    Currency ccy = multicurve.getCurrencyForName(name).get(0);
```

Comment:
If you do know there's only one, then ``Iterators.getOnlyElement(...)`` may be a better approach